BepInCharacterSwapper/Patches/ModelPageManager_Patches.cs

```diff
@@ -1,46 +1,127 @@
-﻿using HarmonyLib;
+﻿using BepInEx.Configuration;
+using HarmonyLib;
+using System.Collections.Generic;
 using System.IO;
 using System.Linq;
 using UnityEngine;
 using UnityEngine.UI;
 
 namespace BepInCharacterSwapper.Patches
 {
-    [HarmonyPatch(typeof(ModelPageManager), "Start")]
+    [HarmonyPatch(typeof(ModelPageManager), nameof(ModelPageManager.Start))]
     internal class ModelPageManager_Patches
     {
         static readonly string PATH = BepInEx.Paths.GameRootPath + "/vrm";
 
+        static List<string> files = new List<string>(); // files to be displayed
+        private static List<string> created = new List<string>(); // buttons created
+
+        private static ModelPageManager? instance;
+
+        private static Dictionary<string, Button> buttonKeypair = new Dictionary<string, Button>(); // button dict (file path, button instance).
+
+        public static void tryGetNewModels()
+        {
+            if (!Directory.Exists(PATH))
+            {
+                Directory.CreateDirectory(PATH);
+                Plugin.Log.LogWarning("vrm folder not found, creating a new one at " + PATH);
+            }
+
+            lock (files)
+            {
+                List<string> remove = new List<string>();
+
+                foreach (string model in files)
+                    if (!File.Exists(model))
+                        remove.Add(model);
+
+                foreach (var item in remove)
+                {
+                    files.Remove(item);
+                    GameObject.Destroy(buttonKeypair[item].gameObject);
+                }
+
+                foreach (string vrmModel in Directory.GetFiles(PATH).Where(x => x.EndsWith(".vrm")))
+                {
+                    if(!files.Contains(vrmModel))
+                        files.Add(vrmModel);
+                }
+            }
+
+            UpdateModelList();
+        }
+
         private static void Postfix(ModelPageManager __instance)
```

Comment:
Is there a way to do this differently? Writing a patch to simply retrieve an instance is somewhat out of scope. If your code needs to be ran multiple times then a helper class should be created that the patch references, such as a "ButtonHelper" or something.